relayer/pkg/chainlink/txm/txm.go

```diff
@@ -250,47 +260,54 @@ func (txm *starktxm) broadcast(ctx context.Context, publicKey *felt.Felt, accoun
 
 	// TODO: consider making this configurable
 	// pad estimate to 250% (add extra because estimate did not include validation)
-	gasConsumed := friEstimate.GasConsumed.BigInt(new(big.Int))
+	gasConsumed := friEstimate.L1GasConsumed.BigInt(new(big.Int))
 	expandedGas := new(big.Int).Mul(gasConsumed, big.NewInt(250))
 	maxGas := new(big.Int).Div(expandedGas, big.NewInt(100))
-	tx.ResourceBounds.L1Gas.MaxAmount = starknetrpc.U64(starknetutils.BigIntToFelt(maxGas).String())
+	broadCastTxnV3.InvokeTxnV3.ResourceBounds.L1Gas.MaxAmount = starknetrpc.U64(starknetutils.BigIntToFelt(maxGas).String())
+
+	L2gasConsumed := friEstimate.L2GasConsumed.BigInt(new(big.Int))
+	L2expandedGas := new(big.Int).Mul(L2gasConsumed, big.NewInt(250))
+	L2maxGas := new(big.Int).Div(L2expandedGas, big.NewInt(100))
+	broadCastTxnV3.InvokeTxnV3.ResourceBounds.L2Gas.MaxAmount = starknetrpc.U64(starknetutils.BigIntToFelt(L2maxGas).String())
 
 	// pad by 150%
-	gasPrice := friEstimate.GasPrice.BigInt(new(big.Int))
+	gasPrice := friEstimate.L1GasPrice.BigInt(new(big.Int))
+	L2gasPrice := friEstimate.L2GasPrice.BigInt(new(big.Int))
 	overallFee := friEstimate.OverallFee.BigInt(new(big.Int)) // overallFee = gas_used*gas_price + data_gas_used*data_gas_price
 
 	// TODO: consider making this configurable
 	// pad estimate to 150% (add extra because estimate did not include validation)
 	gasUnits := new(big.Int).Div(overallFee, gasPrice)
 	expandedGasUnits := new(big.Int).Mul(gasUnits, big.NewInt(150))
 	maxGasUnits := new(big.Int).Div(expandedGasUnits, big.NewInt(100))
-	tx.ResourceBounds.L1Gas.MaxAmount = starknetrpc.U64(starknetutils.BigIntToFelt(maxGasUnits).String())
+	broadCastTxnV3.InvokeTxnV3.ResourceBounds.L1Gas.MaxAmount = starknetrpc.U64(starknetutils.BigIntToFelt(maxGasUnits).String())
```

Comment:
And here, we can remove this calculation for `ResourceBounds.L1Gas.MaxAmount` entirely since we already did this earlier